admin/plugins.php

```diff
@@ -0,0 +1,399 @@
+	 * Filters the plugin card description on the Add Plugins screen.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string $description Plugin card description.
+	 * @param array  $plugin      An array of plugin data. See {@see plugins_api()}
+	 *                            for the list of possible values.
+	 */
+	$description = apply_filters( 'plugin_install_description', $description, $plugin );
+	$version     = $plugin['version'];
+	$name        = wp_strip_all_tags( $title . ' ' . $version );
+	$author      = $plugin['author'];
+	if ( ! empty( $author ) ) {
+		/* translators: %s: Plugin author. */
+		$author = ' <cite>' . sprintf( __( 'By %s', 'default' ), $author ) . '</cite>';
+	}
+
+	$requires_php = isset( $plugin['requires_php'] ) ? $plugin['requires_php'] : null;
+	$requires_wp  = isset( $plugin['requires'] ) ? $plugin['requires'] : null;
+
+	$compatible_php = is_php_version_compatible( $requires_php );
+	$compatible_wp  = is_wp_version_compatible( $requires_wp );
+	$tested_wp      = ( empty( $plugin['tested'] ) || version_compare( get_bloginfo( 'version' ), $plugin['tested'], '<=' ) );
+	$action_links   = array();
+
+	if ( current_user_can( 'install_plugins' ) || current_user_can( 'update_plugins' ) ) {
+		$status = install_plugin_install_status( $plugin );
+
+		switch ( $status['status'] ) {
+			case 'install':
+				if ( $status['url'] ) {
+					if ( $compatible_php && $compatible_wp ) {
+						$action_links[] = sprintf(
+							'<a class="install-now button" data-slug="%s" href="%s" aria-label="%s" data-name="%s">%s</a>',
+							esc_attr( $plugin['slug'] ),
+							esc_url( $status['url'] ),
+							/* translators: %s: Plugin name and version. */
+							esc_attr( sprintf( _x( 'Install %s now', 'plugin', 'default' ), $name ) ),
+							esc_attr( $name ),
+							esc_html__( 'Install Now', 'default' )
+						);
+					} else {
+						$action_links[] = sprintf(
+							'<button type="button" class="button button-disabled" disabled="disabled">%s</button>',
+							esc_html( _x( 'Cannot Install', 'plugin', 'default' ) )
+						);
+					}
+				}
+				break;
+
+			case 'update_available':
+				if ( $status['url'] ) {
+					if ( $compatible_php && $compatible_wp ) {
+						$action_links[] = sprintf(
+							'<a class="button aria-button-if-js" data-plugin="%s" data-slug="%s" href="%s" aria-label="%s" data-name="%s">%s</a>',
+							esc_attr( $status['file'] ),
+							esc_attr( $plugin['slug'] ),
+							esc_url( $status['url'] ),
+							/* translators: %s: Plugin name and version. */
+							esc_attr( sprintf( _x( 'Update %s now', 'plugin', 'default' ), $name ) ),
+							esc_attr( $name ),
+							esc_html__( 'Update Now', 'default' )
+						);
+					} else {
+						$action_links[] = sprintf(
+							'<button type="button" class="button button-disabled" disabled="disabled">%s</button>',
+							esc_html( _x( 'Cannot Update', 'plugin', 'default' ) )
+						);
+					}
+				}
+				break;
+
+			case 'latest_installed':
+			case 'newer_installed':
```

Comment:
I'm not sure we'll want to display the update UI here. Other than that it's a bit of an odd experience since we cannot redirect back to this screen, it's really not important to be able to update the plugins from this screen. We want to focus on installing and activating these plugins, and right now a pending update would actually take precedence over being able to activate the plugin.

So maybe we could just fall through to the regular activation UI? Let's discuss what makes most sense here, but just for illustration, here's what I am thinking:
```suggestion
			case 'update_available':
			case 'latest_installed':
			case 'newer_installed':
```

Reply:
@felixarntz this has been handled in https://github.com/WordPress/performance/pull/864/commits/d471ddeb3fd646b8fa9c312377a2f9d56ea26e87 but I am wondering if we should hint by _some_ means that an update is available, but perhaps that is taking away from what core already does via its plugin UI... I'm happy to leave this as is after applying your suggestion.

Reply:
Thanks @10upsimon. I think it's okay to leave the update UI out for now. We can always readd later if we find there are drawbacks of not having it.